pkg/util/log/format_crdb_v2.go

```diff
@@ -447,3 +455,257 @@ func (buf *buffer) maybeMultiLine(
 var startRedactionMarker = string(redact.StartMarker())
 var endRedactionMarker = string(redact.EndMarker())
 var markersStartWithMultiByteRune = startRedactionMarker[0] >= utf8.RuneSelf && endRedactionMarker[0] >= utf8.RuneSelf
+
+var (
+	entryREV2 = regexp.MustCompile(
+		`(?m)^` +
+			/* Severity                 */ `(?P<severity>[` + severityChar + `])` +
+			/* Date and time            */ `(?P<datetime>\d{6} \d{2}:\d{2}:\d{2}.\d{6}) ` +
+			/* Goroutine ID             */ `(?:(?P<goroutine>\d+) )` +
+			/* Go standard library flag */ `(?P<gostd>\(gostd\) )?` +
+			/* Channel                  */ `(?:(?P<channel>\d+)@)?` +
+			/* File                     */ `(?P<file>[^:]+):` +
+			/* Line                     */ `(?:(?P<line>\d+) )` +
+			/* Redactable flag          */ `(?P<redactable>(?:` + redactableIndicator + `)?) ` +
+			/* Context tags             */ `(?:\[(?P<tags>(?:[^]]|\][^ ])+)\])` +
+			/* Counter                  */ `(?: (?P<counter>\d+))? ? ?` +
```

Comment:
This isn't quite right. Especially the ` ? ?` after counter. What about:

```
			/* Context tags             */ `(?:\[(?P<tags>(?:[^]]|\][^ ])+)\])` +
			/* Counter                  */ `(?: (?P<counter>\d+)|  )?` +
```

The reason I did it like this is that there are some entries that are "non standard" which have just one total space between the tag and the message. Do we need deal with those too? See:

```
I181130 22:14:34.828612 740 storage/store_rebalancer.go:277  [n1,s1,store-rebalancer] load-based lease transfers successfully brought s1 down to 37128.65 qps (mean=29834.62, upperThreshold=37293.27)
```

Reply:
Just noticed there was a bug in the testdata: a log entry that was crdb-v1 formatted. Fixed and I changed the above to: 
```
                       /* Context tags             */ `(?:\[(?P<tags>(?:[^]]|\][^ ])+)\] )` +
                       /* Counter                     */ `(?P<counter>(?:\d+)?) ` +
```

Passes all the tests.